packages/server/src/arpa_reporter/services/validate-upload.js

```diff
@@ -207,18 +207,52 @@ async function validateReportingPeriod({ upload, records, trns }) {
     return errors;
 }
 
+/**
+ * Check whether UEI is required for the recipient
+ * @param {object} recipient - the recipient record
+ * @returns {boolean} - true if UEI is required, false otherwise
+ */
+function isUEIRequired(recipient) {
+    // As of Q1, 2023 we require a UEI for all entities of type subrecipient and/or contractor
```

Comment:
I thought the rule was `As of Q1, 2023 entity of type beneficiary does not require a UEI`. I'm wondering if the logic should be `return recipient.Entity_type_2.doesnotinclude('beneficiary')` instead.

Reply:
🤦  I should have just read the ticket. This logic is correct!

Reply:
Right, I didn't want to call out "non-beneficiary" because in the future if for instance we add more entity types this logic wouldn't hold whereas explicitly checking for one of the two others would